docs/source/migration.mdx

```diff
@@ -270,226 +309,166 @@ The following `ApolloServer` constructor options have been removed in favor of o
-  typeDefs: `
-    type Query { dog: Dog }
-    type Dog { name: String }
-  `,
-  resolvers: {
-    Query: {
-      dog: async (_, __, context) => {
-        return (await context.dataSources.dogs.getDogs())[0];
-      },
-    },
-  },
-});
-
-const { url } = await startStandaloneServer(server, {
-  async context() {
-    // This function is called for every incoming operation, and
-    // returns the context object
-    return {
-      // Create a new instance of each data source for every operation
-      dataSources: {
-        dogs: new DogsDataSource(),
-      },
-    };
-  },
-  listen: { port: 4000 },
-});
-
-console.log(`Server ready at ${url}`);
-```
-
-```js {25-34}
-// Simple example data source
-class DogsDataSource {
-  constructor() {}
-
-  async getDogs() {
-    return [{ name: 'Fido' }];
+interface ContextValue {
+  token: string;
+  dataSources: {
+    moviesAPI: MoviesAPI;
   }
-}
+};
 
 const server = new ApolloServer({
-  typeDefs: `
-    type Query { dog: Dog }
-    type Dog { name: String }
-  `,
-  resolvers: {
-    Query: {
-      dog: async (_, __, context) => {
-        return (await context.dataSources.dogs.getDogs())[0];
-      },
-    },
+  typeDefs,
+  resolvers,
+  context: ({ req: ExpressRequest }): Omit<ContextValue, 'dataSources'> => {
+    return {
+      token: getTokenFromRequest(req),
+    };
   },
-});
-
-const { url } = await startStandaloneServer(server, {
-  async context() {
-    // This function is called for every incoming operation, and
-    // returns the context object
+  dataSources: (): ContextValue['dataSources'] => {
     return {
-      // Create a new instance of each data source for every operation
-      dataSources: {
-        dogs: new DogsDataSource(),
-      },
+      moviesAPI: new MoviesAPI(),
     };
   },
-  listen: { port: 4000 },
 });
 
-console.log(`Server ready at ${url}`);
+await server.listen();
 ```
 
 </MultiCodeBlock>
 
-In the above example, we create a `DogsDataSource` class and then instantiate a new `DogsDataSource` within our `context` initialization function. Now, with every incoming operation, our `context` initialization function returns an object containing a new instance of `DogsDataSource`, accessible to our resolvers from `context.dataSources.dogs`.
+<!-- TODO(AS4): Update the next line before release -->
 
-> For more details about creating `context` in Apollo Server 4, see [`context` initialization function](#context-initialization-function).
+Here's how you would write the same code in Apollo Server 4. (Note that `RESTDataSource` has moved to a new package name (or at least it will [by the time AS4 is released](https://github.com/apollographql/apollo-server/issues/6048)).)
 
-Expand the panel below to see another example of setting up a data source with the `RESTDataSource` class:
+<MultiCodeBlock>
 
-TODO: fully review example
+```ts
+import { RESTDataSource, RESTDataSourceOptions } from '@apollo/datasource-rest';
```

Comment:
I think we should continue using the `apollo-datasource-rest` package in this example until  `@apollo/datasource-rest` is ready to go. Otherwise, this example won't work, leading to confusion for those trying to follow along. 
```suggestion
import { RESTDataSource, RESTDataSourceOptions } from 'apollo-datasource-rest';
```

I'd instead suggest moving the above note: 

> Note that in the future, the `apollo-datasource-rest` package is going to be renamed  `@apollo/datasource-rest` [by the time Apollo Server 4 is released](https://github.com/apollographql/apollo-server/issues/6048)).

to below this snippet, so folks still have the heads up that the package name is _going_ to change soon.

Reply:
I don't think `apollo-datasource-rest` will actually work with AS4 for various technical reasons (basically, it depends on some AS3 stuff that will confuse the TS compiler, and also it expects the `initialize` function to be called). Maybe we should call this out with a big "hey this doesn't quite work yet" at the beginning of the section. We should work on this pretty soon I think. We can link to https://github.com/apollographql/apollo-server/issues/6048 in the callout.